tests/utils/test_events.py

```diff
@@ -360,7 +360,7 @@ def test_NthTradingDayOfWeek(self, n):
     @subtest(param_range(MAX_WEEK_RANGE), 'n')
     def test_NDaysBeforeLastTradingDayOfWeek(self, n):
         cal = get_calendar('NYSE')
-        rule = NDaysBeforeLastTradingDayOfWeek(n)
+        rule = NDaysBeforeLastTradingDayOfWeek(n, True)
```

Comment:
I think `invert` is really tied to the particular class being instantiated here.  What do you think about moving the decision to invert into each subclass's `__init__`?